core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1103,6 +1108,13 @@ public void listStatus(AlluxioURI path, ListStatusContext context,
                   inodePath.getUri());
             }
             if (shouldLoadMetadataIfNotExists(inodePath, loadMetadataContext)) {
+              if (context.isPartialListing() && !context.isPartialListingInitialCall()) {
+                // The path was removed after the first call to the partial listing,
+                // so just throw an exception as the path no longer exists
+                throw new FileDoesNotExistException(
```

Comment:
This can be done at the client since this is just one of the successive calls.

Reply:
But I think the behavior here is to interrupt the operation instead of letting it succeed with tolerance of lost files? I'm a bit concerned that if a dir is frequently updated then this partial list call can have a low success rate, would that happen?

Reply:
Oh this only happens in the case of using the inode id for the listing offset, any will only happen if the inode is deleted or moved. I think most operations will use the "startAfter" option instead of the offset which would be POSIX compatible.

Reply:
I didn't really follow the use cases like the offset and `startAfter` you mention here. But based on my limited understanding it will confuse new comers to this part like me, where you can find many similar options and don't know where each can be from. It would be great if you can best-effort add some comments to how each of them is used. Not a hard requirement for this particular PR :)

Reply:
I added info in the proto file, I can add more docs once matching client operations are added.